geomesa-dynamo/geomesa-dynamodb-datastore/src/main/scala/org/locationtech/geomesa/dynamodb/data/DynamoDBFeatureWriter.scala

```diff
@@ -0,0 +1,137 @@
+
+trait DynamoDBPutter {
+  // Allows impl changes between appending and updating writers
+  protected def dynamoDBPut(t: Table, i: Item): Unit = ???
+}
+
+trait DynamoDBFeatureWriter extends SimpleFeatureWriter with DynamoDBPutter {
+  import org.locationtech.geomesa.utils.geotools.RichSimpleFeatureType._
+  def sft: SimpleFeatureType
+  def table: Table
+
+  private[this] var curFeature: SimpleFeature = null
+
+  val dtgIndex = sft.getDtgIndex.get
+
+  private val encoder = new KryoFeatureSerializer(sft)
+
+  private def serialize(item: Item, attr: AnyRef, desc: AttributeDescriptor) = {
+    import java.{lang => jl}
+    desc.getType.getBinding match {
+      case c if c.equals(classOf[jl.Boolean]) =>
+        item.withBoolean(desc.getLocalName, attr.asInstanceOf[jl.Boolean])
+
+      case c if c.equals(classOf[jl.Integer]) =>
+        item.withInt(desc.getLocalName, attr.asInstanceOf[jl.Integer])
+
+      case c if c.equals(classOf[jl.Double]) =>
+        item.withDouble(desc.getLocalName, attr.asInstanceOf[jl.Double])
+
+      case c if c.equals(classOf[String]) =>
+        item.withString(desc.getLocalName, attr.asInstanceOf[String])
+
+      case c if c.equals(classOf[java.util.Date]) =>
+        item.withLong(desc.getLocalName, attr.asInstanceOf[java.util.Date].getTime)
+
+      case c if classOf[Point].isAssignableFrom(c) =>
+        item.withBinary(desc.getLocalName, WKBUtils.write(attr.asInstanceOf[Point]))
+    }
+  }
+
+  override def hasNext: Boolean = true
+
+  override def next(): SimpleFeature = {
+    curFeature = new ScalaSimpleFeature(UUID.randomUUID().toString, sft)
+    curFeature
+  }
+
+  override def remove(): Unit = throw new NotImplementedError("DynamoDB feature writer is append only")
+
+  override def close(): Unit = {}
+
+  override def getFeatureType: SimpleFeatureType = sft
+
+  override def write(): Unit = {
+    import org.locationtech.geomesa.utils.geotools.Conversions._
+
+    // TODO: is getting the centroid here smart?
+    val geom = curFeature.geometry.getCentroid
+    val x = geom.getX
+    val y = geom.getY
+    val dtg = new DateTime(curFeature.getAttribute(dtgIndex).asInstanceOf[Date])
+
+    // hash key
+    val pk = DynamoPrimaryKey(dtg, x, y)
+    val hash = Ints.toByteArray(pk.idx)
+
+    // range key
+    val secondsInWeek = DynamoPrimaryKey.secondsInCurrentWeek(dtg)
+    val z3 = DynamoPrimaryKey.SFC3D.index(x, y, secondsInWeek)
+    val z3idx = Longs.toByteArray(z3.z)
+
+    val range = Bytes.concat(z3idx, curFeature.getID.getBytes(StandardCharsets.UTF_8))
+
+    //
+    val primaryKey = new PrimaryKey(
+      DynamoDBDataStore.geomesaKeyHash, hash,
+      DynamoDBDataStore.geomesaKeyRange, range
+    )
+
+    val item = new Item().withPrimaryKey(primaryKey)
+
+    curFeature.getAttributes.zip(sft.getAttributeDescriptors).foreach { case (attr, desc) => serialize(item, attr, desc) }
+    item.withBinary(DynamoDBDataStore.serId, encoder.serialize(curFeature))
```

Comment:
we are including the other attributes so that we can eventually either filter/query on attributes and or reconstruct the simple features types from attributes of the item (for now kryo serialization will suffice). I will be creating jira tickets to capture these points before the FCR pr is up.